lib/utils.js

```diff
@@ -212,6 +215,91 @@ const retryWithCheck = async (method, check, maxAttempts = 5, delayInMillisecond
   return result;
 };
 
+/**
+ * 
+ * @param {BtcTransactionHelper} btcTxHelper 
+ * @returns {Promise<Array<string>>} the mempool tx ids
+ */
+const getBitcoinMempool = async (btcTxHelper) => {
+  const mempool = await btcTxHelper.nodeClient.execute('getrawmempool', []);
+  return mempool;
+};
+
+/**
+ * 
+ * @param {BtcTransactionHelper} btcTxHelper 
+ * @param {Promise<string>} btcTxHash 
+ */
+const waitForBitcoinTxToBeInMempool = async (btcTxHelper, btcTxHash) => {
+
+  const bitcoinMempoolHasTx = async () => {
+    const bitcoinMempool = await getBitcoinMempool(btcTxHelper);
+    const txIsInMempool = bitcoinMempool.includes(btcTxHash);
+    if(!txIsInMempool) {
+      console.debug(`Attempting to check if the btc tx (${btcTxHash}) was already mined since it's not in the mempool yet.`);
+      const tx = await btcTransactionHelper.getTransaction(btcTxHash);
+      if(tx) {
+        console.debug(`The btc tx (${btcTxHash}) was already mined.`);
+        return true;
+      }
+      return false;
+    }
+    return true;
+  };
+
+  const checkBitcoinMempoolHasTx = async (btcTxAlreadyFound, currentAttempts) => {
+    if(btcTxAlreadyFound) {
+      console.debug(`The btc tx ${btcTxHash} was found in the mempool at attempt ${currentAttempts}.`);
+    } else {
+      console.log(`Attempting to get the btc tx ${btcTxHash} in the mempool. Attempt: ${currentAttempts}.`);
+    }
+    return btcTxAlreadyFound;
+  };
+
+  const onError = async (e) => {
+    if(e.message.includes('No such mempool or blockchain transaction')) {
+      console.debug(`The btc tx ${btcTxHash} is not in the mempool nor mined yet. Let's allow some more time before retrying to get it.`);
+      return true;
+    }
+    console.error(`Un expected error while trying to get the btc tx ${btcTxHash} in the mempool.`, e);
+    throw e;
+  };
+
+  await retryWithCheck(bitcoinMempoolHasTx, checkBitcoinMempoolHasTx, 5, 1000, onError);
+};
+
+/**
+ * Waits until the bitcoin mempool has at least one tx.
+ * @param {BtcTransactionHelper} btcTxHelper 
+ * @returns {Promise<boolean>}
+ */
+const waitForBitcoinMempoolToGetTxs = async (btcTxHelper, maxAttempts = 5, checkEveryMilliseconds = 1000) => {
+
+  const initialBitcoinMempoolSize = (await getBitcoinMempool(btcTxHelper)).length;
+
+  console.debug(`[waitForBitcoinMempoolToGetTxs] The initial bitcoin mempool size is ${initialBitcoinMempoolSize}.`);
+  console.debug(`Will wait and attempt to check if the bitcoin mempool has received any new transactions ${maxAttempts} times.`);
+
+  const getBitcoinMempoolSize = async () => {
+    const bitcoinMempool = await getBitcoinMempool(btcTxHelper);
+    const bitcoinMempoolSize = bitcoinMempool.length;
+    return bitcoinMempoolSize;
+  };
+
+  const checkBtcMempoolIsNotEmpty = async (bitcoinMempoolSize, currentAttempts) => {
```

Comment:
```suggestion
  const checkBtcMempoolIsNotEmpty = async (bitcoinMempoolSize) => {
```

Reply:
`currentAttempts` is not being used. Is it needed?

Reply:
Updated!